src/v0/destinations/optimizely_fullstack/transform.js

```diff
@@ -28,30 +28,35 @@ const responseBuilder = (payload, endpoint, destination) => {
 };
 
 const trackResponseBuilder = (message, destination) => {
-  const { event } = message;
+  const { event, anonymousId } = message;
   if (!event) {
     throw new InstrumentationError('Event name is required');
   }
-  const { baseUrl } = destination.Config;
+  const { baseUrl, trackKnownUsers } = destination.Config;
   const endpoint = getTrackEndPoint(baseUrl, event);
   const { name } = CONFIG_CATEGORIES.TRACK;
   const payload = constructPayload(message, MAPPING_CONFIG[name]);
+  if (!trackKnownUsers) {
+    payload.userId = anonymousId;
+  }
   return responseBuilder(payload, endpoint, destination);
 };
 
 const pageResponseBuilder = (message, destination) => {
-  let event;
-  if (message.category && destination.Config.trackCategorizedPages) {
-    event = `Viewed ${message.category} page`;
+  const { trackCategorizedPages, trackNamedPages } = destination.Config;
+  const newMessage = { ...message };
+  const returnValue = [];
+
+  if (message.category && trackCategorizedPages) {
+    newMessage.event = `Viewed ${message.category} page`;
+    returnValue.push(trackResponseBuilder(newMessage, destination));
   }
-  if (message.name && destination.Config.trackNamedPages) {
-    event = `Viewed ${message.name} page`;
+  if (message.name && trackNamedPages) {
```

Comment:
does that mean, ultimately the clause `message.name && trackNamedPages` takes precedence over `message.category && trackCategorizedPages`? because, if both are there, the last value overrides the first value of event. If that is the case, then please add comments regarding it, along with ref

Reply:
got it.